### PR TITLE
Some minor refactoring as as we move towards supporting cloudstorage …

### PR DESCRIPTION
**Pull request by mhoak**
…directly.

### CODE REVIEW

File: katalyze/main.go [hunk 1]
```diff
@@ -41,6 +41,8 @@ import (
 
 var (
 	outputDir       = flag.String("output_dir", "", "Directory for returning the processed SGFs. By default, uses current directory")
+	bucket          = flag.String("bucket", "", "The bucket for CloudStorage of analyzed games. This param is optional.")
+	bucketPrefix    = flag.String("bucket_prefix", "", "Dictates the path within the bucket where games wil be stored. Required if the bucket parameter is supplied.")
```
Comment (artasparks) [inline katalyze/main.go:46]: Not sure I understand bucket prefix?

File: katalyze/main.go [hunk 2]
```diff
@@ -41,6 +41,8 @@ import (
 
 var (
 	outputDir       = flag.String("output_dir", "", "Directory for returning the processed SGFs. By default, uses current directory")
+	bucket          = flag.String("bucket", "", "The bucket for CloudStorage of analyzed games. This param is optional.")
```
Comment (artasparks) [inline katalyze/main.go:45]: seems like you need a flag that's like `--storage-provider={localdisk|gcp}`

File: katalyze/main.go (hunk 1 above)
Comment (mhoak) [reply katalyze/main.go:46]: At least for AWS, you usually specify the bucket and the path within the bucket separately. Might be different for GCP

File: katalyze/main.go (hunk 2 above)
Comment (mhoak) [reply katalyze/main.go:45]: Good point. I was considering that. That does seem nicer.

File: katalyze/main.go (hunk 1 above)
Comment (artasparks) [reply katalyze/main.go:46]: Yeah, that's true for GCS also. maybe root folder?

Comment (artasparks) [reply katalyze/main.go:46]: Or maybe prefix is canonical

Comment (mhoak) [reply katalyze/main.go:46]: I think prefix is canonical

**Review by artasparks — APPROVED** [review]

**Comment by TamaBrian** [comment]
Merry Christmas Message

From

    Brian and Etsuko Willis

**Pull request merged**